balena/models/application.py

```diff
@@ -382,24 +382,24 @@ def create(self, name, device_type, organization, app_type=None):
                 raise exceptions.OrganizationNotFound(organization)
 
         device_types = self.config.get_device_types()
-        device_manifest = [device for device in device_types if device['name'] == device_type]
+        device_manifest = [device for device in device_types if (device['name'] == device_type) or (device_type in device.get('aliases', []))]
```

Comment:
It seems that we already didn't support DT aliases, in which case I would highly suggest to replace `config.get_device_types()` with a query to the `/v6/device_type` resource filtering on the `slug $or name`.

I'm already working on adding the device type aliases to the model, so we can start using them once that is merged.
See: https://github.com/balena-io/open-balena-api/pull/898

To be clear that's part of the effort of reducing our dependence and dropping the deprecated `/device-types/v1` endpoint.

```suggestion
```

Reply:
@thgreasi, I know we're changing to the new endpoint but since the latest SDK uses `/device-types/v1` (https://github.com/balena-io/balena-sdk/blob/master/lib/models/config.ts#L158) so I still want to keep using it here and will make a change when the JS SDK updated.

Reply:
@nghiant2710 the only reason that the JS SDK still uses it is in order to support aliases.
Given that the python SDK doesn't support aliases, I don't see any reason use it just for the shake of using it.
I will leave this up to you though, since we can do this change in a follow up.

Reply:
@thgreasi, I just realise that the output of `/v6/device_type` endpoint contains less information than current `/device-types/v1` which potentially breaks other methods so it's better to do this change in another PR.